Use `cloud-init` to enable `lxd-agent` on Ubuntu releases before `20.04` (#12680)
canonical/lxd: doc/reference/devices_disk.md

      lxc config device add <instance_name> <device_name> disk source=cloud-init:config

>   Adding such configuration disk might be needed if the VM image used includes `cloud-init` but not the `lxd-agent`. This is the case of official Ubuntu images prior to `20.04`. On such images, the following steps will enable the LXD agent and thus provide the ability to `lxc exec` into the VM:

Reviewer: ```suggestion
  Adding such a configuration disk might be needed if the VM image that is used includes `cloud-init` but not the `lxd-agent`. This is the case for official Ubuntu images prior to `20.04`. On such images, the following steps enable the LXD agent and thus provide the ability to use `lxc exec` to access the VM:
```

      lxc config device add <instance_name> <device_name> disk source=cloud-init:config

  Adding such a configuration disk might be needed if the VM image that is used includes `cloud-init` but not the `lxd-agent`. This is the case for official Ubuntu images prior to `20.04`. On such images, the following steps enable the LXD agent and thus provide the ability to use `lxc exec` to access the VM:

    lxc init ubuntu-daily:18.04 --vm u1
    lxc config device add u1 config disk source=cloud-init:config
    lxc config set u1 cloud-init.user-data - << EOF
    #cloud-config
    #packages:
    #  - linux-image-virtual-hwe-16.04  # 16.04 GA kernel as a problem with vsock
    runcmd:
      - mount -t 9p config /mnt
      - cd /mnt
      - ./install.sh
      - cd /
      - umount /mnt
      - systemctl start lxd-agent  # XXX: causes a reboot
    EOF
    lxc start --console u1

>   Note that for `16.04`, the HWE kernel is required to workaround a problem with `vsock`, see the commented out section in the above `cloud-config`.

Reviewer: ```suggestion
  Note that for `16.04`, the HWE kernel is required to work around a problem with `vsock` (see the commented out section in the above `cloud-config`).
```
